### PR TITLE
test: nRF53 using icmsg as default IPC transport

### PR DESCRIPTION
**Pull request by hubertmis**
This patch is intended only for test purpose. It shall not be merged to the main branch.

Signed-off-by: Hubert Miś <hubert.mis@nordicsemi.no>

### CODE REVIEW

**Comment by e-rk** [comment]
There might be a race condition and a bug in `spsc_pbuf`
https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/os/spsc_pbuf.c#L254
The variable `rd_idx_loc` holds an address to the read index. In certain configurations, when the value pointed by `rd_idx_loc` is read, some garbage value is obtained and stored into `rd_idx`. The index is then used to read some memory offset, causing a fault.

**Comment by hubertmis** [comment]
@e-rk, have you observed such issue?

File: samples/bluetooth/nrf_dm/child_image/hci_rpmsg.conf [hunk 1]
```diff
@@ -17,8 +17,12 @@ CONFIG_MPSL_TIMESLOT_SESSION_COUNT=1
...
 
+# System workqueue
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+
 # BT
 CONFIG_BT_EXT_ADV=y
 
 # IPC
-CONFIG_IPC_SERVICE_BACKEND_RPMSG_NUM_ENDPOINTS_PER_INSTANCE=3
+CONFIG_IPC_SERVICE_BACKEND_ICMSG_ME_NUM_EP=3
```
Comment (e-rk) [inline samples/bluetooth/nrf_dm/child_image/hci_rpmsg.conf:27]: I think `IPC_SERVICE_BACKEND_ICMSG_ME_NUM_EP` could be merged together with `IPC_SERVICE_BACKEND_RPMSG_NUM_ENDPOINTS_PER_INSTANCE`. There is absolutely no reason why the same configuration from the user point of view should have different names depending on the backend.

**Comment by hubertmis** [comment]
Stale PR

**Pull request closed without merging**